tests/testthat/test-resume.R

```diff
@@ -1,41 +1,147 @@
-  first_phase <- run_resumable_simulation(initial_timesteps, parameters)
+  set.seed(123)
+  first_phase <- run_resumable_simulation(warmup_timesteps, warmup_parameters)
   second_phase <- run_resumable_simulation(
-    total_timesteps,
+    timesteps,
     parameters,
     initial_state=first_phase$state,
     restore_random_state=TRUE)
 
-  set.seed(1)
-  uninterrupted_run <- run_simulation(total_timesteps, parameters=parameters)
+  expect_equal(nrow(first_phase$data), warmup_timesteps)
+  expect_equal(nrow(second_phase$data), timesteps - warmup_timesteps)
+
+  expect_mapequal(
+    second_phase$data,
+    uninterrupted_run[-(1:warmup_timesteps),])
+
+  invisible(second_phase$data)
+}
+
+test_that('Simulation can be resumed', {
+  test_resume(get_parameters(overrides=list(
+    individual_mosquitoes = FALSE,
+    model_seasonality = TRUE
+  )))
+  test_resume(get_parameters(overrides=list(
+    individual_mosquitoes = TRUE,
+    model_seasonality = TRUE
+  )))
+  test_resume(get_parameters(overrides=list(
+    individual_mosquitoes = FALSE,
+    model_seasonality = TRUE
+  )))
+  test_resume(get_parameters(overrides=list(
+    individual_mosquitoes = TRUE,
+    model_seasonality = TRUE
+  )))
+})
+
+test_that('PEV intervention can be added when resuming', {
+  set_default_mass_pev <- function(parameters, timesteps) {
+    n <- length(timesteps)
+    set_mass_pev(
+      parameters,
+      profile = rtss_profile,
+      timesteps = timesteps,
+      coverages = rep(0.5, n),
+      min_wait = 5,
+      min_ages = 365*10,
+      max_ages = 365*60,
+      booster_spacing = NULL,
+      booster_coverage = NULL,
+      booster_profile = NULL)
+  }
 
-  expect_equal(nrow(first_phase$data), initial_timesteps)
-  expect_equal(nrow(second_phase$data), total_timesteps - initial_timesteps)
-  expect_equal(rbind(first_phase$data, second_phase$data), uninterrupted_run)
+  base <- get_parameters(overrides=list(pev_doses=c(0, 45, 90)))
+  data <- test_resume(
+    warmup_parameters = base,
+    parameters = base %>% set_default_mass_pev(100))
+
+  expect_equal(data[data$n_pev_mass_dose_1 > 0, "timestep"], 100)
+  expect_equal(data[data$n_pev_mass_dose_2 > 0, "timestep"], 145)
+  expect_equal(data[data$n_pev_mass_dose_3 > 0, "timestep"], 190)
+
+  # Add a second mass PEV intervention when resuming the simulation.
+  data <- test_resume(
+    warmup_parameters = base %>% set_default_mass_pev(25),
+    parameters = base %>% set_default_mass_pev(c(25, 100)))
+
+  # The first dose, at time step 25, happens during the warmup and is not
+  # returned by test_resume, hence why we don't see it in the data. Follow-up
+  # doses do show up, event though they we scheduled during warmup.
+  expect_equal(data[data$n_pev_mass_dose_1 > 0, "timestep"], c(100))
+  expect_equal(data[data$n_pev_mass_dose_2 > 0, "timestep"], c(70, 145))
+  expect_equal(data[data$n_pev_mass_dose_3 > 0, "timestep"], c(115, 190))
+})
+
+test_that("TBV intervention can be added when resuming", {
```

Comment:
This test seems different from the others, should we name it differently? or remove the TBV parameterisation from the warmup?

```suggestion
test_that("TBV intervention parameters can changed when resuming", {
```

Reply:
I've tweaked the layout of the test a little bit which should make it easier to spot the two parts of it.